services/QuillLMS/app/helpers/navigation_helper.rb

```diff
@@ -115,4 +188,130 @@ def should_show_admin_access_tab?
   def admin_impersonating_user?(user)
     session[:admin_id].present? && session[:admin_id] != user.id
   end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
```

Comment:
I added this disable block because I wasn't really sure if there was a better way to approach this portion of the code that didn't feel like over-engineering-- I'm open to suggestions though

Reply:
I think in some ideal world we probably want to rationalize our paths so that the sections are sort of named in the path and are thus easy and predictable to extract, but I think that's a project that's way bigger than the scope of this PR.

I don't know that this is necessarily better, but it at least avoids a bunch of `if/elsif` statements:
```ruby
ACTIVE_TAB_PATHS = {
  LEARNING_TOOLS => ['tools'],
  ABOUT_US => ['about', 'announcements', 'mission', 'impact', 'press', 'team', 'pathways', 'careers']
  ...
}

def determine_active_tab(current_path)
  ACTIVE_TAB_PATHS.each do |tab, paths|
    # return early once we find our match
    return tab if paths.any? { |path| current_path.include?(path) }
  end
  determine_dashboard_active_tab(current_path)
end
```

Reply:
Agreed with Thomas here - the easiest thing would be to turn the groupings into hashes and iterate. Totally up to you whether that feels worthwhile, though.